graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt

```diff
@@ -276,8 +278,11 @@ class DgsSchemaProvider(
 
     private fun createBasicDataFetcher(method: Method, dgsComponent: Any): DataFetcher<Any?> {
         return DataFetcher<Any?> { environment ->
-            val result = invokeDataFetcher(method, dgsComponent, DgsDataFetchingEnvironment(environment))
-            result
+            when (val result = invokeDataFetcher(method, dgsComponent, DgsDataFetchingEnvironment(environment))) {
```

Comment:
Won't this create a hard dependency on `reactor` ?

Reply:
It would, which I also wasn't happy with.

I've refactored it behind a `DataFetcherResultProcessor` abstraction now, which is implemented in the reactive module: https://github.com/Netflix/dgs-framework/pull/389/commits/395c0a46c19354afa557ea9345b36fff3f4caec3